# 1. Cleaning

### importing data 

In [63]:
import pandas as pd
import openpyxl

In [224]:
csv_file = "D:\\ironhack\\proyectos\\SharkAttack_proy1\DATA\\attacks_1.csv"
shark_attack = pd.read_csv(csv_file, encoding="cp1252")

In [225]:
shark_attack

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
25718         NaN          NaN     NaN         NaN        NaN   
25719         NaN          NaN     NaN         NaN        NaN   
25720         NaN          NaN     NaN         NaN        NaN   
25721         NaN          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
25718              NaN                             NaN          NaN   
25719              NaN                             NaN          NaN   
25720              NaN                             NaN          NaN   
25721              NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
25718              NaN  NaN  ...              NaN   
25719              NaN  NaN  ...              NaN   
25720              NaN  NaN  ...              NaN   
25721              NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
25718                             NaN                       NaN   
25719                             NaN                       NaN   
25720                             NaN                       NaN   
25721                             NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
25718                                                NaN   
25719                                                NaN   
25720                                                NaN   
25721                                                NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [226]:
shark_attack.sample(3)

Case Number Date  Year Type Country Area Location Activity Name Sex   \
17712         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
13194         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
19473         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   

       ... Species  Investigator or Source  pdf href formula href  \
17712  ...      NaN                    NaN  NaN          NaN  NaN   
13194  ...      NaN                    NaN  NaN          NaN  NaN   
19473  ...      NaN                    NaN  NaN          NaN  NaN   

      Case Number.1 Case Number.2 original order Unnamed: 22 Unnamed: 23  
17712           NaN           NaN            NaN         NaN         NaN  
13194           NaN           NaN            NaN         NaN         NaN  
19473           NaN           NaN            NaN         NaN         NaN  

[3 rows x 24 columns]

In [227]:
shark_attack.shape[0]

25723

In [228]:
shark_attack.dtypes

Case Number                object
Date                       object
Year                      float64
Type                       object
Country                    object
Area                       object
Location                   object
Activity                   object
Name                       object
Sex                        object
Age                        object
Injury                     object
Fatal (Y/N)                object
Time                       object
Species                    object
Investigator or Source     object
pdf                        object
href formula               object
href                       object
Case Number.1              object
Case Number.2              object
original order            float64
Unnamed: 22                object
Unnamed: 23                object
dtype: object

### Null values

In [229]:
shark_attack.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25723 entries, 0 to 25722
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             8702 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 3464 non-null 

In [230]:
#revisar si necesito eliminar los NaN
#puedo meterme directo a explorar la columna a ver como se ven con los NaN
#llamando el key y poniendole .isna()

In [231]:
#pd.isnull(shark_attack).sum() same formula
shark_attack.isna().sum()
#for NaN, un promedio de 18000NaNs en una lista de 19803
#conclucion, demasiados NaN tendre que eliminar

Case Number               17021
Date                      19421
Year                      19423
Type                      19425
Country                   19471
Area                      19876
Location                  19961
Activity                  19965
Name                      19631
Sex                       19986
Age                       22252
Injury                    19449
Fatal (Y/N)               19960
Time                      22775
Species                   22259
Investigator or Source    19438
pdf                       19421
href formula              19422
href                      19421
Case Number.1             19421
Case Number.2             19421
original order            19414
Unnamed: 22               25722
Unnamed: 23               25721
dtype: int64

In [232]:
shark_attack.notnull().sum()

Case Number               8702
Date                      6302
Year                      6300
Type                      6298
Country                   6252
Area                      5847
Location                  5762
Activity                  5758
Name                      6092
Sex                       5737
Age                       3471
Injury                    6274
Fatal (Y/N)               5763
Time                      2948
Species                   3464
Investigator or Source    6285
pdf                       6302
href formula              6301
href                      6302
Case Number.1             6302
Case Number.2             6302
original order            6309
Unnamed: 22                  1
Unnamed: 23                  2
dtype: int64

After this filter we can see that it has tons of null data, from 25723 original rows we can see that it will let us with almos 6,000 rows. Not much information and we can clean it.


next step it will be to drop (errase) the rows that dont apport nothing to the data. 
If we throw the formula od dropna with the fuction "any" its not an option, every single row has even one missed cell, we will be applying the function "all"

In [233]:
shark_attack.dropna(how="all",inplace=True)
shark_attack

Case Number         Date    Year        Type    Country  \
0      2018.06.25  25-Jun-2018  2018.0     Boating        USA   
1      2018.06.18  18-Jun-2018  2018.0  Unprovoked        USA   
2      2018.06.09  09-Jun-2018  2018.0     Invalid        USA   
3      2018.06.08  08-Jun-2018  2018.0  Unprovoked  AUSTRALIA   
4      2018.06.04  04-Jun-2018  2018.0    Provoked     MEXICO   
...           ...          ...     ...         ...        ...   
8698            0          NaN     NaN         NaN        NaN   
8699            0          NaN     NaN         NaN        NaN   
8700            0          NaN     NaN         NaN        NaN   
8701            0          NaN     NaN         NaN        NaN   
25722          xx          NaN     NaN         NaN        NaN   

                  Area                        Location     Activity  \
0           California     Oceanside, San Diego County     Paddling   
1              Georgia  St. Simon Island, Glynn County     Standing   
2               Hawaii                    Habush, Oahu      Surfing   
3      New South Wales              Arrawarra Headland      Surfing   
4               Colima                        La Ticla  Free diving   
...                ...                             ...          ...   
8698               NaN                             NaN          NaN   
8699               NaN                             NaN          NaN   
8700               NaN                             NaN          NaN   
8701               NaN                             NaN          NaN   
25722              NaN                             NaN          NaN   

                  Name Sex   ...         Species   \
0          Julie Wolfe    F  ...      White shark   
1      Adyson McNeely     F  ...              NaN   
2          John Denges    M  ...              NaN   
3                 male    M  ...        2 m shark   
4       Gustavo Ramos     M  ...  Tiger shark, 3m   
...                ...  ...  ...              ...   
8698               NaN  NaN  ...              NaN   
8699               NaN  NaN  ...              NaN   
8700               NaN  NaN  ...              NaN   
8701               NaN  NaN  ...              NaN   
25722              NaN  NaN  ...              NaN   

               Investigator or Source                       pdf  \
0                    R. Collier, GSAF      2018.06.25-Wolfe.pdf   
1      K.McMurray, TrackingSharks.com    2018.06.18-McNeely.pdf   
2      K.McMurray, TrackingSharks.com     2018.06.09-Denges.pdf   
3                      B. Myatt, GSAF  2018.06.08-Arrawarra.pdf   
4                           A .Kipper      2018.06.04-Ramos.pdf   
...                               ...                       ...   
8698                              NaN                       NaN   
8699                              NaN                       NaN   
8700                              NaN                       NaN   
8701                              NaN                       NaN   
25722                             NaN                       NaN   

                                            href formula  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...   
1      http://sharkattackfile.net/spreadsheets/pdf_di...   
2      http://sharkattackfile.net/spreadsheets/pdf_di...   
3      http://sharkattackfile.net/spreadsheets/pdf_di...   
4      http://sharkattackfile.net/spreadsheets/pdf_di...   
...                                                  ...   
8698                                                 NaN   
8699                                                 NaN   
8700                                                 NaN   
8701                                                 NaN   
25722                                                NaN   

                                                    href Case Number.1  \
0      http://sharkattackfile.net/spreadsheets/pdf_di...    2018.06.25   
1      http://sharkattackfile.net/spreadsheets/pdf_di...    2

In [234]:
shark_attack.shape[0]

8703

it let us with only 8703 rows. But we still have some missing values in the cells thats wy apears in the formula below that we still have a lot of them, probably half of the columns have NANs.

In [235]:
shark_attack.isna().sum()

Case Number                  1
Date                      2401
Year                      2403
Type                      2405
Country                   2451
Area                      2856
Location                  2941
Activity                  2945
Name                      2611
Sex                       2966
Age                       5232
Injury                    2429
Fatal (Y/N)               2940
Time                      5755
Species                   5239
Investigator or Source    2418
pdf                       2401
href formula              2402
href                      2401
Case Number.1             2401
Case Number.2             2401
original order            2394
Unnamed: 22               8702
Unnamed: 23               8701
dtype: int64

### Duplicated

In [236]:
shark_attack.shape[0]

8703

In [237]:
shark_attack.duplicated().sum()

2392

This tell us that we have 2,392 raws that are duplicated and will have to be removed, otherwise it will leave us with some false information.

In [238]:
shark_attack.drop_duplicates(inplace=True)
shark_attack.sample(5)

Case Number                      Date    Year        Type  \
3694  1964.02.14.a               14-Feb-1964  1964.0  Unprovoked   
6170  1580.01.10.R  Letter dated 10-Jan-1580  1580.0  Unprovoked   
4586    1947.02.00                  Feb-1947  1947.0  Unprovoked   
4852    1937.03.26               26-Mar-1937  1937.0     Invalid   
441     2015.03.18               18-Mar-2015  2015.0  Unprovoked   

                       Country             Area  \
3694                      FIJI       Vanua Levu   
6170  Between PORTUGAL & INDIA              NaN   
4586                    BRAZIL   Rio de Janeiro   
4852                 AUSTRALIA  New South Wales   
441                        USA           Hawaii   

                                Location  \
3694  Nailou Village, Tunuloa Natewa Bay   
6170                                 NaN   
4586                          Copacabana   
4852                      South Cronulla   
441                         Hapuna Beach   

                                               Activity         Name Sex   \
3694                                       Spearfishing   Ivo Berabi    M   
6170  Man fell overboard from ship. Those on board t...         male    M   
4586                                            Bathing          NaN  NaN   
4852                                            Bathing  Austin Shaw    M   
441                               Standing / Snorkeling  Ken Grasing    M   

      ...                                           Species   \
3694  ...                                                NaN   
6170  ...                                                NaN   
4586  ...                                                NaN   
4852  ...  Shark involvement prior to death was not confi...   
441   ...                          Tiger shark, 8 to 12 feet   

              Investigator or Source                              pdf  \
3694          Fiji Times; L. Schultz          1964.02.14.a-Berabi.pdf   
6170             G.P. Whitley, p. 10  1580.01.10.R-Portugal-India.pdf   
4586                 Globo, 5/7/1997            1947.02.00-Brazil.pdf   
4852  Canberra Times, 3/29/1937, p.1              1937.03.26-Shaw.pdf   
441                  KMBC, 3/19/2015           2015.03.18-Grasing.pdf   

                                           href formula  \
3694  http://sharkattackfile.net/spreadsheets/pdf_di...   
6170  http://sharkattackfile.net/spreadsheets/pdf_di...   
4586  http://sharkattackfile.net/spreadsheets/pdf_di...   
4852  http://sharkattackfile.net/spreadsheets/pdf_di...   
441   http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
3694  http://sharkattackfile.net/spreadsheets/pdf_di...  1964.02.14.a   
6170  http://sharkattackfile.net/spreadsheets/pdf_di...  1580.01.10.R   
4586  http://sharkattackfile.net/spreadsheets/pdf_di...    1947.02.00   
4852  http://sharkattackfile.net/spreadsheets/pdf_di...    1937.03.26   
441   http://sharkattackfile.net/spreadsheets/pdf_di...    2015.03.18   

     Case Number.2 original order Unnamed: 22 Unnamed: 23  
3694  1964.02.14.a         2609.0         NaN         NaN  
6170  1580.01.10.R          133.0         NaN         NaN  
4586    1947.02.00         1717.0         NaN         NaN  
4852    1937.03.26         1451.0         NaN         NaN  
441     2015.03.18         5862.0         NaN         NaN  

[5 rows x 24 columns]

In [239]:
shark_attack.shape[0]

6311

This leave us with only 6,311 rows that are not duplicated and have minimum 1 real value.
Because I can notice that there are still some rows that looks preatty empty and the few values that could have it might not be important to us and our investigation.

### columns

passing throw the Data frame I notice that theres some columns that are not relevant to our investigation, that doesnt apport anything of value that we could could benefit

In [240]:
shark_attack.sample()

Case Number         Date    Year     Type Country     Area  \
1597  2005.04.09  09-Apr-2005  2005.0  Invalid     USA  Florida   

                                 Location Activity             Name Sex   ...  \
1597  Central Gulf Coast, St. John County  Surfing  Glenn Henderson    M  ...   

                             Species   \
1597  Shark involvement not confirmed   

                                 Investigator or Source  \
1597  WTSP TampaBays10.com; First Coast News, April ...   

                           pdf  \
1597  2005.04.09-Henderson.pdf   

                                           href formula  \
1597  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
1597  http://sharkattackfile.net/spreadsheets/pdf_di...    2005.04.09   

     Case Number.2 original order Unnamed: 22 Unnamed: 23  
1597    2005.04.09         4706.0         NaN         NaN  

[1 rows x 24 columns]

In [241]:
shark_attack.columns

Index(['Case Number', 'Date', 'Year', 'Type', 'Country', 'Area', 'Location',
       'Activity', 'Name', 'Sex ', 'Age', 'Injury', 'Fatal (Y/N)', 'Time',
       'Species ', 'Investigator or Source', 'pdf', 'href formula', 'href',
       'Case Number.1', 'Case Number.2', 'original order', 'Unnamed: 22',
       'Unnamed: 23'],
      dtype='object')

There are some columns that makes me have some interest, in wich ones we could errase because of our lak of information for us, for example the las 2 columns 'Unnamed: 22', 'Unnamed: 23', seen the samples we could see that theres only non values and actually the tittle doesn´t give us any explanetion.
We will erase them.

In [242]:
shark_attack[['Unnamed: 22', "Unnamed: 23"]].sample(5)

Unnamed: 22 Unnamed: 23
279          NaN         NaN
4372         NaN         NaN
2566         NaN         NaN
5309         NaN         NaN
6066         NaN         NaN

In [243]:
shark_attack.drop(columns =['Unnamed: 22', "Unnamed: 23"], inplace=True, axis=1)

In [244]:
shark_attack.sample()

Case Number         Date    Year        Type           Country  \
4234  1956.08.25  25-Aug-1956  1956.0  Unprovoked  PAPUA NEW GUINEA   

                  Area                                       Location  \
4234  Central Province  Paga Point or Fishermans Island, Port Moresby   

      Activity                    Name Sex   ... Fatal (Y/N) Time  \
4234  Fishing   Kara Benagi, from Hula    M  ...           Y  NaN   

               Species                              Investigator or Source  \
4234  4.3 m [14'] shark  V.M. Coppleson (1958), pp.49-51 & 264; A.M. Ra...   

                        pdf  \
4234  1956.08.25-Benagi.pdf   

                                           href formula  \
4234  http://sharkattackfile.net/spreadsheets/pdf_di...   

                                                   href Case Number.1  \
4234  http://sharkattackfile.net/spreadsheets/pdf_di...    1956.08.25   

     Case Number.2 original order  
4234    1956.08.25         2069.0  

[1 rows x 22 columns]

more research in columns: lets see if the internet links could help us with our research.

In [245]:
shark_attack['href formula'].sample(5)                        

5317    http://sharkattackfile.net/spreadsheets/pdf_di...
472     http://sharkattackfile.net/spreadsheets/pdf_di...
5390    http://sharkattackfile.net/spreadsheets/pdf_di...
2711    http://sharkattackfile.net/spreadsheets/pdf_di...
4055    http://sharkattackfile.net/spreadsheets/pdf_di...
Name: href formula, dtype: object

In [246]:
shark_attack['href'].sample(5) 

2489    http://sharkattackfile.net/spreadsheets/pdf_di...
3826    http://sharkattackfile.net/spreadsheets/pdf_di...
6043    http://sharkattackfile.net/spreadsheets/pdf_di...
4701    http://sharkattackfile.net/spreadsheets/pdf_di...
4839    http://sharkattackfile.net/spreadsheets/pdf_di...
Name: href, dtype: object

Exploring this 2 columns 'href formula' and 'href' we can realice that they apporr some internet links, but going throw them they the pages dont exist amymore.

In [247]:
shark_attack['Case Number.1'].sample(5) 

3206        1975.10.12
5613    1897.03.15.a.R
3091        1979.08.26
2630        1990.07.08
5621      1896.09.11.R
Name: Case Number.1, dtype: object

other columns that dont apport anything of value are 'Case Number.1', 'Case Number.2'  we can see that the information there is actually the same and its the date of the attack that we do have it already in other columns at the beginning, and 'original order' dont aport us anything.
we´ll pass to some more drop cleaning.

In [248]:
shark_attack.drop(columns =['href formula', 'href', 'Case Number.1', 'Case Number.2', 'original order'], inplace=True, axis=1)

In [249]:
shark_attack.sample()

Case Number         Date    Year        Type Country               Area  \
1220  2008.08.16  16-Aug-2008  2008.0  Unprovoked     USA  US Virgin Islands   

         Location        Activity             Name Sex  Age  \
1220  Buck Island  Treading water  Elizabeth Riggs    F  38   

                               Injury Fatal (Y/N)  Time  \
1220  Severe lacerations to left foot           N  Dusk   

                                   Species  Investigator or Source  \
1220  8' bull shark or Caribbean reef shark         M. Levne, GSAF   

                       pdf  
1220  2008.08.16-Riggs.pdf

In [250]:
shark_attack.columns = [i.capitalize().replace(" ", "_") for i in list(shark_attack.columns)]
shark_attack.columns

Index(['Case_number', 'Date', 'Year', 'Type', 'Country', 'Area', 'Location',
       'Activity', 'Name', 'Sex_', 'Age', 'Injury', 'Fatal_(y/n)', 'Time',
       'Species_', 'Investigator_or_source', 'Pdf'],
      dtype='object')

In [251]:
#shark_attack.columns = [i.replace(" ", "") for i in list(shark_attack.columns) if i[-1]==" "]
#shark_attack.columns

In [252]:
#for i in list(shark_attack.columns):
 #   i[-1].replace(" ", "")
#shark_attack.columns

In [253]:
#for i in list(shark_attack.columns):
#    for e in range(len(i)):
#        if e == "_":
#            list(shark_attack.columns).append(e.replace("_", ""))
#shark_attack.columns

In [254]:
#for i in list(shark_attack.columns):
#    for e in len(i)if i[-1] == " ":
#        list(shark_attack.columns).append(i.replace(" ", ""))
#shark_attack.columns

In [255]:
shark_attack.sample(50)

Case_number                       Date    Year          Type  \
3522     1967.07.05                05-Jul-1967  1967.0    Unprovoked   
3585     1966.03.12                14-Mar-1966  1966.0  Sea Disaster   
4767     1940.03.20                20-Mar-1940  1940.0    Unprovoked   
21     2018.04.25.a                25-Apr-2018  2018.0    Unprovoked   
3334     1972.05.06                06-May-1972  1972.0    Unprovoked   
5787   1883.02.25.R       Reported 25-Feb-1883  1883.0    Unprovoked   
102      2017.08.10                10-Aug-2017  2017.0    Unprovoked   
3062     1980.07.00              Late Jul-1980  1980.0    Unprovoked   
3653     1964.12.06                06-Dec-1964  1964.0    Unprovoked   
1125     2009.05.17                17-May-2009  2009.0    Unprovoked   
699      2013.03.03                03-Mar-2013  2013.0    Unprovoked   
3573     1966.06.22                22-Jun-1966  1966.0      Provoked   
626    2013.09.07.a                07-Sep-2013  2013.0    Unprovoked   
5741     1887.09.06                11-Sep-1887  1887.0       Invalid   
1913     2001.08.27                27-Aug-2001  2001.0    Unprovoked   
1123     2009.05.31                31-May-2009  2009.0      Provoked   
3171     1976.09.17                17-Sep-1976  1976.0  Sea Disaster   
1102     2009.07.31               31-July-2009  2009.0    Unprovoked   
5302   1916.07.13.b                13-Jul-1916  1916.0       Invalid   
5873   1875.11.27.R       Reported 27-Nov-1875  1875.0    Unprovoked   
5494     1905.07.02                10-Jul-1905  1905.0    Unprovoked   
486      2014.10.17                17-Oct-2014  2014.0    Unprovoked   
5194     1923.12.02                02-Dec-1923  1923.0    Unprovoked   
3641     1965.01.09                09-Jan-1965  1965.0    Unprovoked   
5775  1884.08.28.R.       Reported 28-Aug-1884  1884.0    Unprovoked   
2754     1988.01.14                14-Jan-1988  1988.0      Provoked   
1183     2008.12.20                20-Dec-2008  2008.0       Boating   
3193     1976.01.12                12-Jan-1976  1976.0    Unprovoked   
1045     2010.02.03                03-Feb-2010  2010.0    Unprovoked   
5102     1928.07.00              Late Jul-1928  1928.0      Provoked   
5091   1929.00.00.c                       1929  1929.0      Provoked   
1127   2009.05.16.a                16-May-2009  2009.0    Unprovoked   
4256     1956.03.25                25-Mar-1956  1956.0       Boating   
5333     1914.09.09                09-Sep-1914  1914.0    Unprovoked   
233      2016.08.04                04-Aug-2016  2016.0    Unprovoked   
3414   1970.00.00.b                      1970s  1970.0    Unprovoked   
5527   1902.12.22.R       Reported 22-Dec-1902  1902.0    Unprovoked   
1569   2005.07.15.R  Reported      15-Jul-2005  2005.0       Invalid   
1095     2009.08.30                30-Aug-2009  2009.0    Unprovoked   
1949     2001.05.04                04-May-2001  2001.0    Unprovoked   
973    2010.10.28.R       Reported 28-Oct-2010  2010.0    Unprovoked   
947      2011.02.12                12-Feb-2011  2011.0    Unprovoked   
6244        ND.0062                Before 1960     0.0    Unprovoked   
3981   1960.07.02.a                02-Jul-1960  1960.0    Unprovoked   
2463     1993.11.21                21-Nov-1993  1993.0    Unprovoked   
4260     1956.02.26                26-Feb-1956  1956.0    Unprovoked   
1578     2005.06.16                16-Jun-2005  2005.0    Unprovoked   
5356   1913.08.26.R                26-Aug-1913  1913.0       Invalid   
2873   1984.10.21.b                21-Oct-1984  1984.0       Invalid   
1926     2001.08.05                05-Aug-2001  2001.0    Unprovoked   

               Country                                      Area  \
3522            TURKEY                            Mugla Province   
3585             SUDAN                                   Red Sea   
4767         AUSTRALIA                           New South Wales   
21           AUSTRALIA                         Western Australi

In [261]:
shark_attack['Sex_'].value_counts()

M      5094
F       637
M         2
N         2
lli       1
.         1
Name: Sex_, dtype: int64

In [256]:
shark_attack["Type"].value_counts()

Unprovoked      4595
Provoked         574
Invalid          547
Sea Disaster     239
Boating          203
Boat             137
Questionable       2
Boatomg            1
Name: Type, dtype: int64

In [258]:
shark_attack['Sex_'].value_counts()

M      5094
F       637
M         2
N         2
lli       1
.         1
Name: Sex_, dtype: int64

In [51]:
shark_attack['Age'].value_counts()

17                154
18                150
19                142
20                141
15                139
                 ... 
20?                 1
 28                 1
7      &    31      1
 30                 1
13 or 14            1
Name: Age, Length: 157, dtype: int64

In [260]:
shark_attack['Species_'].value_counts()

White shark                                                                                                                      163
Shark involvement prior to death was not confirmed                                                                               105
Invalid                                                                                                                          102
Shark involvement not confirmed                                                                                                   88
Tiger shark                                                                                                                       73
                                                                                                                                ... 
1.2 m to 1.5 m [4.5' to 5'] shark                                                                                                  1
Bull shark, 2.3 m [7.5']                                             

In [ ]:
shark_attack['Species_'].value_counts()

In [45]:
#no proprciona mucha informacion, o nula.
shark_attack.describe()

Year
count  6300.000000
mean   1927.272381
std     281.116308
min       0.000000
25%    1942.000000
50%    1977.000000
75%    2005.000000
max    2018.000000

In [46]:
shark_attack.sample()

Case Number         Date    Year        Type Country        Area  \
3995  1960.04.24  24-Apr-1960  1960.0  Unprovoked     USA  California   

                         Location                                 Activity  \
3995  Tomales Point, Marin County  Skindiving for abalone (but at surface)   

                  Name Sex  Age                   Injury Fatal (Y/N)   Time  \
3995  Frank I. Gilbert    M  48  Foot & swim fin bitten            N  14h00   

                       Species   \
3995  White shark, 4.9 m [16']    

                                 Investigator or Source  \
3995  F.I. Gilbert; Follett, p. 192-198; D. Miller &...   

                                 pdf  
3995  1960.04.24-Gilbert_Collier.pdf

In [47]:
shark_attack["Country"].value_counts()

USA                       2229
AUSTRALIA                 1338
SOUTH AFRICA               579
PAPUA NEW GUINEA           134
NEW ZEALAND                128
                          ... 
MALDIVE ISLANDS              1
NICARAGUA                    1
NORTH SEA                    1
RED SEA / INDIAN OCEAN       1
CEYLON (SRI LANKA)           1
Name: Country, Length: 212, dtype: int64